components/merino/src/curated_recommendations/mod.rs

```diff
@@ -0,0 +1,377 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+mod error;
+mod http;
+mod models;
+
+pub use error::{ApiResult, Error, Result};
+use error_support::handle_error;
+pub use models::{CuratedRecommendationsRequest, CuratedRecommendationsResponse};
+
+#[derive(uniffi::Object)]
+pub struct CuratedRecommendationsClient {
+    inner: CuratedRecommendationsClientInner<http::HttpClient>,
+    base_host: String,
+    user_agent_header: String,
+}
+
+struct CuratedRecommendationsClientInner<T: http::HttpClientTrait> {
+    http_client: T,
+}
+
+#[uniffi::export]
+impl CuratedRecommendationsClient {
+    #[uniffi::constructor]
+    #[handle_error(Error)]
+    pub fn new(base_host: String, user_agent_header: String) -> ApiResult<Self> {
```

Comment:
I just realized that while `user_agent_header` should be required, `base_host` is better left as optional with the default being the production host. It's very likely that we will add more arguments like these in the future, so let's use the builder pattern for the client creation here. You can [reference](https://github.com/mozilla/application-services/blob/main/components/suggest/src/store.rs#L38-L128) this as an example.